File: count-pairs-in-two-arrays/count-pairs-in-two-arrays.cpp
class Solution {
public:
    long long countPairs(vector<int>& nums1, vector<int>& nums2) {
        vector<int> diff(nums1.size());
        
        for (int i = 0; i < nums1.size(); i++) 
            diff[i] = nums1[i] - nums2[i];
        
        sort(diff.begin(), diff.end());
        
        long long ans = 0;
        
        int l = 0, r = nums1.size() - 1;
        
        while (l < r) {
            const int ll = abs(diff[l]);
            const int rr = abs(diff[r]);
            if (rr > ll || diff[l] > 0) {
                ans += r - l;
                r--;
            } 
            else {
                l++;
            }
        }
        return ans;
    }
};

File: 2036-count-pairs-in-two-arrays/count-pairs-in-two-arrays.py
class Solution:
    def countPairs(self, nums1: List[int], nums2: List[int]) -> int:
        arr = []
        for num1, num2 in zip(nums1,nums2):
            arr.append(num1-num2)
        arr.sort()
        ans = 0
        n = len(arr)
        for i in range(n):
            if arr[i] > 0:
                ans += n - i - 1
            else:
                j = bisect_right(arr, -arr[i], i + 1, n)
                ans += n - j
        return ans